server/src/main/java/org/opensearch/rest/action/admin/cluster/RestClusterGetSettingsAction.java

```diff
@@ -122,4 +128,19 @@ static ClusterGetSettingsResponse response(
         );
     }
 
+    /**
+     * Parse the deprecated request parameter 'master_timeout', and add deprecated log if the parameter is used.
+     * It also validates whether the value of 'master_timeout' is the same with 'cluster_manager_timeout'.
+     * Remove the method along with MASTER_ROLE.
+     * @deprecated As of 2.0, because promoting inclusive language.
+     */
+    @Deprecated
+    private static void parseDeprecatedMasterTimeoutParameter(ClusterStateRequest clusterStateRequest, RestRequest request) {
```

Comment:
Can we refactor this as a public static method in `BaseRestHandler.java` and abstract it to take a base `MasterNodeRequest` instance so we don't have to duplicate in each of these files?

e.g., 

```java
    /**
     * Parse the deprecated request parameter 'master_timeout', and add deprecated log if the parameter is used.
     * It also validates whether the value of 'master_timeout' is the same with 'cluster_manager_timeout'.
     * Remove the method along with MASTER_ROLE.
     * @deprecated As of 2.0, because promoting inclusive language.
     */
    @Deprecated
    public static void parseDeprecatedMasterTimeoutParameter(MasterNodeRequest mnr, RestRequest request) {
        final String deprecatedTimeoutParam = "master_timeout";
        if (request.hasParam(deprecatedTimeoutParam)) {
            deprecationLogger.deprecate("cluster_state_master_timeout_parameter", MASTER_TIMEOUT_DEPRECATED_MESSAGE);
            request.validateParamValuesAreEqual(deprecatedTimeoutParam, "cluster_manager_timeout");
            mnr.masterNodeTimeout(request.paramAsTime(deprecatedTimeoutParam, mnr.masterNodeTimeout()));
        }
    }
```

Reply:
Hi Nick, thanks a lot for your time in the review!
Sure, I will make the suggested change. I realized adding the duplicate method into every `Action` class is a pain, as well as removing in the future, and my idea was avoid adding the API-specific method into the abstract base class.
Since it's a temporary method and you agree with it, I will move it into `BaseRestHandler` class. 😁

Reply:
ty!

Reply:
@nknize I put your suggested change into a separate PR https://github.com/opensearch-project/OpenSearch/pull/2670, and I plan to take that PR as a foundation, then deprecate `master_timeout` parameter in the other REST APIs (such as PR https://github.com/opensearch-project/OpenSearch/pull/2557 & https://github.com/opensearch-project/OpenSearch/pull/2658)

Reply:
The duplidated `parseDeprecatedMasterTimeoutParameter()` method is put into abstract base class `BaseRestHandler` through PR #2670, and I have updated this PR. 😁